src/xnnpack/prefetch.h

```diff
@@ -6,7 +6,37 @@
 #pragma once
 
 #ifdef _MSC_VER
-  #include <intrin.h>
+  #include <intrin.h>  
+#endif
+
+#ifdef __hexagon__
+  #include <hexagon_protos.h>
+
+  // Use fetchl2 at least several hundred cycles prior to using the data.
+  
+  // height: the number of 128 bytes to fetch.
+  // width: we set 128 bytes here.
+  // stride: we set 128 bytes here.
+  XNN_INLINE static void xnn_prefetch_to_l2_linear(const void *address, uint32_t height)
+  {
+    uint32_t info = ((128 << 16) | ((128 << 8) & 0xffff) | (height & 0xffff));
+    Q6_l2fetch_AR(address, info);
+  }
+
+  // height: the number of width-sized blocks to fetch.
+  // width: width of a fetch block in bytes.
+  // stride: an unsigned byte offset which is used to increment the pointer
+  //         after each width-sized block is fetched.
+  // direction: If clear, row major. If set, cloumn major.
+  XNN_INLINE static void xnn_prefetch_to_l2_box(const void *address,
+                                        uint64_t stride,
+                                        uint64_t width,
+                                        uint64_t height,
+                                        uint64_t direction)
+  {
+    uint64_t info = ((h3 << 48) | ((h2 & 0xffff) << 32) | ((h1 & 0xffff) << 16) | (h0 & 0xffff));
```

Comment:
Where do h0, h1, h2, h3 come from?

Reply:
Fixed, it was wrong paste from tester code.